plugins/org.jkiss.dbeaver.ext.vertica/src/org/jkiss/dbeaver/ext/vertica/edit/VerticaConstraintManager.java

```diff
@@ -0,0 +1,89 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.vertica.edit;
+
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.ext.generic.edit.GenericPrimaryKeyManager;
+import org.jkiss.dbeaver.ext.generic.model.GenericUniqueKey;
+import org.jkiss.dbeaver.ext.vertica.model.VerticaConstraint;
+import org.jkiss.dbeaver.model.DBConstants;
+import org.jkiss.dbeaver.model.DBPEvaluationContext;
+import org.jkiss.dbeaver.model.DBUtils;
+import org.jkiss.dbeaver.model.edit.DBEPersistAction;
+import org.jkiss.dbeaver.model.exec.DBCExecutionContext;
+import org.jkiss.dbeaver.model.impl.edit.DBECommandAbstract;
+import org.jkiss.dbeaver.model.impl.edit.SQLDatabasePersistAction;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.model.sql.SQLUtils;
+import org.jkiss.dbeaver.model.struct.DBSEntityConstraintType;
+import org.jkiss.utils.CommonUtils;
+
+import java.util.List;
+import java.util.Map;
+
+public class VerticaConstraintManager extends GenericPrimaryKeyManager {
+
+    @Override
+    protected void addObjectCreateActions(DBRProgressMonitor monitor, DBCExecutionContext executionContext, List<DBEPersistAction> actions, ObjectCreateCommand command, Map<String, Object> options) {
+        VerticaConstraint constraint = (VerticaConstraint) command.getObject();
+        if (command.getObject().getConstraintType() == DBSEntityConstraintType.CHECK) {
+            actions.add(
+                    new SQLDatabasePersistAction(
+                            "Create check constraint",
+                            "ALTER TABLE " + constraint.getParentObject().getFullyQualifiedName(DBPEvaluationContext.DDL) +
+                                    " ADD CONSTRAINT CHECK (" + constraint.getCheckConstraintDefinition() + ")"
+                    ));
+        } else {
+            super.addObjectCreateActions(monitor, executionContext, actions, command, options);
+        }
+    }
+
+    @Override
+    protected void addObjectModifyActions(DBRProgressMonitor monitor, DBCExecutionContext executionContext, List<DBEPersistAction> actionList, ObjectChangeCommand command, Map<String, Object> options) throws DBException {
+        VerticaConstraint constraint = (VerticaConstraint) command.getObject();
+
+        if (command.getProperties().containsKey(DBConstants.PROP_ID_ENABLED)) {
+            actionList.add(
+                    new SQLDatabasePersistAction("Alter constraint",
+                            "ALTER TABLE " + constraint.getTable().getFullyQualifiedName(DBPEvaluationContext.DDL) +
+                                    " ALTER CONSTRAINT " + DBUtils.getQuotedIdentifier(constraint.getDataSource(), constraint.getName()) + " " +
+                                    (constraint.isEnabled() ? "ENABLED" : "DISABLED")
+                    )
+            );
```

Comment:
Speaking of indentation: we use 4 spaces for _continuous_ indentation.

In IntelliJ IDEA head to <kbd>Editor</kbd> &rArr; <kbd>Code Style</kbd> &rArr; <kbd>Java</kbd> &rArr; <kbd>Continuation indent</kbd> and change the value from `8` to `4`.